.github/workflows/main.yml

```
@@ -14,158 +11,64 @@ on:
 
 jobs:
   build:
-    if: github.event_name == 'pull_request'
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-
-      - uses: actions/setup-go@v5
-        with:
-          go-version-file: 'go.mod'
-
-      - name: Generate protobuf
-        run: make pb
-
-      - name: Build the BESS-UPF Docker image
-        run: DOCKER_TARGETS=bess make docker-build
-
-      - name: Build the PFCP Agent Docker image
-        run: DOCKER_TARGETS=pfcpiface make docker-build
+    uses: "omec-project/.github/.github/workflows/upf-build-upf.yml@main"
+    with:
+      branch_name: ${{ github.ref }}
 
   build-ptf:
-    if: github.event_name == 'pull_request'
-    runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ./ptf
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Build PTF image
-        run: make build
+    uses: "omec-project/.github/.github/workflows/upf-build-ptf.yml@main"
```

This task takes a lot of time, and this is why we want it to run only for `pull_requests`
```suggestion
    if: github.event_name == 'pull_request'
    uses: "omec-project/.github/.github/workflows/upf-build-upf.yml@main"
```